enarx-keep/src/backend/probe/x86_64.rs

```diff
@@ -3,30 +3,76 @@
 use crate::backend::Datum;
 
 use std::arch::x86_64::{CpuidResult, __cpuid_count};
-use std::convert::Into;
+use std::convert::TryInto;
+use std::io::{Error, ErrorKind, Result};
+use std::mem::transmute;
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum Vendor {
+    Amd,
+    Intel,
+}
+
+impl Vendor {
+    pub fn get() -> Result<Self> {
+        let res = unsafe { __cpuid_count(0x00000000, 0x00000000) };
+        let name: [u8; 12] = unsafe { transmute([res.ebx, res.edx, res.ecx]) };
+        let name = std::str::from_utf8(&name[..]).unwrap();
+
+        Ok(match name {
+            "AuthenticAMD" => Self::Amd,
+            "GenuineIntel" => Self::Intel,
+            _ => {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    format!("unsupported vendor: {}", name),
+                ))
+            }
+        })
+    }
+}
 
 pub struct CpuId {
     pub name: &'static str,
     pub leaf: u32,
     pub subl: u32,
     pub func: fn(CpuidResult) -> (bool, Option<String>),
+    pub vend: Option<Vendor>,
+}
+
+impl CpuId {
+    pub fn fail(&self) -> Datum {
+        Datum {
+            name: self.name.into(),
+            pass: false,
+            info: None,
+            mesg: None,
+        }
+    }
 }
 
-impl Into<Datum> for &CpuId {
-    fn into(self) -> Datum {
-        let max = unsafe { __cpuid_count(0x00000000, 0x00000000) }.eax;
+impl TryInto<Datum> for &CpuId {
```

Comment:
You need `TryFrom` here instead. You basically never impl `Into` or `TryInto` in Rust.

Reply:
On second thought, don't make this a fallible conversion. Make it an infallible conversion and merge `fail()` into the `cpu vendor mismatch` case.

Reply:
Addressed in latest revision